src/Shared/Components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,169 @@
+import { ButtonHTMLAttributes, ChangeEvent, useCallback, useEffect, useState } from 'react'
+import { AnimatePresence, motion } from 'framer-motion'
+import { CustomInput, useRegisterShortcut, UseRegisterShortcutProvider } from '@Common/index'
+import { ComponentSizeType } from '@Shared/constants'
+import { ConfirmationModalProps } from './types'
+import { getPrimaryButtonStyleFromVariant, getConfirmationLabel, getIconFromVariant } from './utils'
+import { Button, ButtonStyleType, ButtonVariantType } from '../Button'
+import './confirmationModal.scss'
+import { Backdrop } from '../Backdrop'
+
+let timeoutId: number
+
+const ConfirmationModal = ({
+    title,
+    subtitle,
+    Icon,
+    variant,
+    buttonConfig,
+    customInputConfig,
+    children,
+    showConfirmationModal,
+    handleClose,
+}: ConfirmationModalProps) => {
+    const { registerShortcut, unregisterShortcut } = useRegisterShortcut()
+
+    const [confirmationText, setConfirmationText] = useState<string>('')
+
+    const customInputIdentifier = customInputConfig?.identifier
+    const confirmationKeyword = customInputConfig?.confirmationKeyword
+
+    const { primaryButtonConfig, secondaryButtonConfig } = buttonConfig
+
+    const RenderIcon = Icon ?? getIconFromVariant(variant)
+
+    const disablePrimaryButton: boolean = confirmationKeyword && confirmationText.trim() !== confirmationKeyword
+
+    const handleTriggerPrimaryActionButton = () => {
+        if (primaryButtonConfig && !disablePrimaryButton) {
+            primaryButtonConfig.onClick()
+        }
+    }
+
+    const handleEnterKeyPress = (e: KeyboardEvent) => {
+        if (e.key === 'Enter') {
+            handleTriggerPrimaryActionButton()
+        }
+    }
+
+    const handleCloseWrapper = useCallback(() => {
+        if (!primaryButtonConfig?.isLoading && !secondaryButtonConfig?.disabled) {
+            handleClose()
+        }
+    }, [primaryButtonConfig, secondaryButtonConfig])
+
+    useEffect(() => {
+        if (showConfirmationModal) {
+            // Timeout so that if modal is opened on enter press, it does not trigger onClick
+            timeoutId = setTimeout(() => {
+                registerShortcut({ keys: ['Enter'], callback: handleTriggerPrimaryActionButton })
+            }, 100)
+        }
+
+        return () => {
+            if (showConfirmationModal) {
+                unregisterShortcut(['Enter'])
+                clearTimeout(timeoutId)
```

Comment:
This should be outside this check ideally check should be if (timeoutId) { clearTimeout(timeoutId) }